File: sample/src/main/java/com/fondesa/lyra/sample/model/AutoSaveEditText.java
package com.fondesa.lyra.sample.model;

import android.content.Context;
import android.os.Parcelable;
import android.support.v7.widget.AppCompatEditText;
import android.util.AttributeSet;

import com.fondesa.lyra.Lyra;
import com.fondesa.lyra.annotation.SaveState;

public class AutoSaveEditText extends AppCompatEditText {
    @SaveState
    CharSequence mText;

    public AutoSaveEditText(Context context) {
        super(context);
    }

    public AutoSaveEditText(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public AutoSaveEditText(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public Parcelable onSaveInstanceState() {
        mText = getText();
        return Lyra.instance().saveState(this, super.onSaveInstanceState());
    }

    @Override
    public void onRestoreInstanceState(Parcelable state) {
        super.onRestoreInstanceState(Lyra.instance().restoreState(this, state));
        setText(mText);
    }
}

File: sample/src/main/kotlin/com/fondesa/lyra/sample/model/AutoSaveEditText.kt
package com.fondesa.lyra.sample.model

import android.content.Context
import android.os.Parcelable
import android.support.v7.widget.AppCompatEditText
import android.util.AttributeSet

import com.fondesa.lyra.Lyra
import com.fondesa.lyra.annotation.SaveState

class AutoSaveEditText : AppCompatEditText {
    @SaveState
    internal var insertedText: CharSequence? = null

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun onSaveInstanceState(): Parcelable {
        insertedText = text
        return Lyra.instance().saveState(this, super.onSaveInstanceState())
    }

    override fun onRestoreInstanceState(state: Parcelable) {
        super.onRestoreInstanceState(Lyra.instance().restoreState(this, state))
        setText(insertedText)
    }
}